src/Compiler/SyntaxTree/XmlDoc.fs

```diff
@@ -76,13 +76,15 @@ type XmlDoc(unprocessedLines: string[], range: range) =
             match paramNamesOpt with
             | None -> ()
             | Some paramNames ->
+                let paramsSet = paramNames |> Set.ofList
```

Comment:
Same here, we're allocating additional set for every xmldoc. What would memory and gc implications be?

Reply:
Same here, the for over xml.Descendants and then List.contains is 3x O(n * n) where over a Set it's 3x O(n * log n)

Reply:
Hmm, as far as I understand, this code is reading external library XML doc file `<param>` tags. I would expect library either users xml param tags (C# library) and there are a lot of those, or doesn't use (F# `///` typically use only summary tags) and then there are no params.

Reply:
Out of general interest I built a Benchmark test. I couldn't make it real because the types have so much internal constructs, and I don't know the exact match-rate vs size of objects, but I simulated similar code structure, and behold:

| Method   | Mean     | Error    | StdDev   | Gen0   | Allocated |
|--------- |---------:|---------:|---------:|-------:|----------:|
| ListTest | 11.89 ns | 0.169 ns | 0.141 ns | 0.0032 |      40 B |
| SetTest  | 27.55 ns | 0.293 ns | 0.274 ns | 0.0095 |     120 B |

Meanwhile logically thinking Set would make sense, it seems in real life `|> Set.ofList` is removing all the performance benefits, probably because it performs some kind of distinct-operation.